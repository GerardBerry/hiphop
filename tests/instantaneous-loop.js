"use strict"

var rjs = require("hiphop");

var prg = <rjs.ReactiveMachine debug name="instloop">
  <rjs.outputsignal name="O" type="string"/>
  <rjs.loop>
    <rjs.localsignal name="L" type="string">
      <rjs.emit signal_name="L" exprs=${"foo bar"}/>
      <rjs.emit signal_name="O" exprs=${rjs.value("L")}/>
    </rjs.localsignal>
  </rjs.loop>
</rjs.ReactiveMachine>;

try {
   prg.react();
   prg.react();
   prg.react();
   prg.react();
} catch (e) {
   console.log(e.message)
}
