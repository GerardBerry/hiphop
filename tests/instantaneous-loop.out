*** RUNTIME ERROR
*** /home/colin/phd/hiphop/tests/instantaneous-loop.js:119
*** Instantaneous loop on incarnation 0.
